### DISCUSSION TITLE
Button name and value stripped from form params when using x-bind

### DISCUSSION THREAD

**Discussion by msolli** [alpinejs/alpine#4098] [1. Help] [answered]
Hi all

I'm having weird problem where the name and value of my submit button isn't included in the form params when submitting. The problem manifests when there's an Alpine x-bind on the button's disabled attribute:
```html
<form action="https://echo.free.beeceptor.com/" method="post" x-data="{ submitting: false }" @submit="submitting = true">
  <button type="submit" name="_action" value="publish" :disabled="submitting">
    <span x-show="! submitting">Publish</span>
    <span x-show="submitting" style="display: none;">Publishing…</span>
  </button>
</form>
```

I want the button's name and value to be submitted along with the form, as in vanilla HTML, but here it doesn't. I'm trying to understand what is causing this, and whether it might be a bug or some undocumented behavior in Alpine.

Here's a Codepen to illustrate the problem: https://codepen.io/msolli/pen/poBNXJj

**Comment by SimoTod** [comment, marked as answer]
you are changing `submitting` to true when you submit the form which adds the disabled attribute to the button. Disabled inputs and buttons are not included in a form request.

You are trying to give to much responsibility to that button, imo.
Just change it to something like
```html
<form action="https://echo.free.beeceptor.com/" method="post" x-data="{ submitting: false }" @submit="submitting = true">
  <input type="hidden" name="_action" value="publish">
  <button type="submit" :disabled="submitting">
    <span x-show="! submitting">Publish</span>
    <span x-show="submitting" style="display: none;">Publishing…</span>
  </button>
</form>
```

**Reply by msolli** [reply]
Ah yes! 🤦 Thanks!

**Comment by ekwoka** [comment]
Alternatively, you can "disable" the button without using the disabled attribute, but just making it unclickable

**Reply by docdunning** [reply]
I sometimes have more than one submit button on a form, and use the button's value so that the server can tell which action to take.  In these cases I would probably hide all the buttons when the form is submitted, and toggle in a 'Please wait ...' message.  I've also seen some cases where the message is a dummy button.

```
  <form action="https://echo.free.beeceptor.com" method="post" x-data="{ submitting: false }" @submit="submitting = true">
    <button x-show="! submitting" type="submit" name="_action" value="publish">Publish</button>
    <button x-show="submitting" type="button" disabled>Publishing ...</button>
  </form>

```

**Reply by SimoTod** [reply]
> I sometimes have more than one submit button on a form, and use the button's value so that the server can tell which action to take

In theory, you can submit the form by just pressing enter on a text input. If you have multiple actions for the same form, and a user presses enter, it will just send one value but there's no guarantee it will send the action your user wanted to perform. Something to bear in mind. 👍

**Reply by docdunning** [reply]
> In theory, you can submit the form by just pressing enter on a text input

@SimoTod - that's a very good point.  To be honest, I can't remember how I avoided that, but like you say, it's definitely something to watch out for.
